File: LATIHAN_KHUSUS/Fungsi_Struktur_data_Percabangan.js
// Struktur data

var data = [
     "Halo Dunia",
     "Halo World",
     "Halo Fets",
     "Halo Jundy",
     "Halo Gundy",
     "Halo Rundy",
]

for (a = 0; a < data.length; a++) {
     console.log (data [a])
}


console.log ("\n --- batas --- \n")




// Array 1

var fer = [
     "Halo Dunk",
     "Halo Dri",
     "Halo dun",
     "Halo Der",
     "Halo Wer",
     "Halo Qwe",
     "Halo Ban",
     "Halo Dun"
]

for (b = 0; b < fer.length; b++) {
     console.log (fer [b])
}


console.log ("\n --- batas --- \n")



// For dasar

var der = [
     "Halo Just",
     "Halo DUst",
     "Halo Ampe",
     "Halo Guntime",
     "Halo Goodtime",
]

for (c = 0; c < der.length; c++) {
     console.log (der [c])
}


console.log ("\n --- batas --- \n")



// Array + Manipulasi data

var fer =  [
     "Halo China",
     "Halo Taiwan",
     "Halo Thailand",
     "Halo Malaysia",
     "Halo Indonesia",
     "Halo Brunei",
     "Halo Timor",
     "Halo Papua"
]

fer.push ("Halo Kamboja")
fer.push ("Halo Yen")
fer.push ("Halo Fer")
fer.push ("Halo Guntime")
fer.push ("Halo Der")


for (f of fer) {
     if (f == "Halo Fer") {
          continue
     }

     console.log (f)
}

console.log ("\n --- batas --- \n")




// Array 3

var ter = [
     "Halo Inkong",
     "Halo Rankong",
     "Halo Bust",
     "Halo Runway",
]

for (f = 0; f < ter.length; f++) {
     console.log (ter [f])
}


console.log ("\n --- batas --- \n")




// Fungsi dasar

function dasar () {
     console.log ("Hello World")
}

dasar ()


console.log ("\n --- batas --- \n")




// Function dengan parameter

function nama (sapa) {
     console.log (`Halo saya ${sapa} dari Jakarta Utara`)
}

nama ("Sapa")
nama ("Hayyan")
nama ("Vayyan")
nama ("Yun")


console.log ("\n --- batas --- \n")




// Fungsi return

function dek (sapa) {
     return `Halo saya ${sapa} dari Jakarta Utara`
}

console.log (dek ("Hayyan"))
console.log (dek ("Fayyan"))
console.log (dek ("Fayyan"))
console.log (dek ("Ron"))
console.log (dek ("jon"))
console.log (dek ("Koper"))



console.log ("\n --- batas --- \n")




// Fungsi return dengan operator dasar

function tambah (x, y) {
     return x + y
}


function kurang (x, y) {
     return x - y
}


function kali (x, y) {
     return x * y
}


function bagi (x, y) {
     return x / y
}


console.log ("Hasil tambah =", tambah (10, 3))
console.log ("Hasil kurang =", kurang (10, 5))
console.log ("Hasil kali =", kali (10, 10))
console.log ("Hasil bagi =", bagi (10, 2))


console.log ("\n --- batas --- \n")



// Error Handling

try {
     var a = 10 / mon
     console.log (a)
}

catch (Error) {
     console.log ("Gagal")
}

finally {
     console.log ("Selesai")
}


console.log ("\n --- batas --- \n")



// Error Handling

try {
     var b = 10 + 10
     console.log (b)
}

catch (Error) {
     console.log ("Gagal")
}

finally {
     console.log ("Selesai")
}


console.log ("\n --- batas --- \n")




// Fungsi dengan percabangan

function hal (d) {

     if (d >= 5) {
          console.log (`Angka d besar, angka d = ${d}`)
     }

     else {
          console.log (`Angka d kecil, angka d = ${d}`)
     }
}

hal (10)
hal (8)
hal (3)
hal (2)
hal (1)


console.log ("\n --- batas --- \n")

File: LATIHAN_KHUSUS/Fungsi_Struktur_data_Percabangan.py
# Struktur data

daf = [
     "Halo Dunia",
     "Halo World",
     "Halo Fets",
     "Halo Jundy",
     "Halo Gundy",
     "Halo Rundy",
]

for j in daf:
     print (j)


print ("\n --- batas --- \n")



# Array 1

fer = [
     "Halo Dunk",
     "Halo Dri",
     "Halo dun",
     "Halo Der",
     "Halo Wer",
     "Halo Qwe",
     "Halo Ban",
     "Halo Dun"
]

for u in fer:
     print (u)


print ("\n --- batas --- \n")




# Array 3

ger = [
     "Halo Der",
     "Halo Dun",
     "Halo Zhong",
     "Halo Wei",
     "Halo Er Wei",
     "Halo Der",
     "Halo Fer", 
]

ger.append ("Halo hun")
ger.append ("Halo der")
ger.append ("Halo run")
ger.append ("Halo fer")
ger.append ("Halo Fen")
ger.append ("Halo Der")
ger.append ("Halo Dun")

for j in ger:
     if j == "Halo Fer":
          continue
     print (j)


print ("\n --- batas --- \n")




# Array + For

gun = [
     "Halo Dun",
     "Halo Lip",
     "Halo Der",
     "Halo dun",
     "halo dun",
     "afer",
     "fert",
     "gert"
]

gun.append ("Halo Dunf")
gun.append ("Halo Der")
gun.append ("Halo Dek")
gun.append ("Halo Det")
gun.append ("Halo Qwe")

for i in gun:
     if i == "Gert":
          continue
     print (i)


print ("\n --- batas --- \n")




# For end 

fet = [
     "Halo Dun",
     "Sky",
     "Hun",
     "Hunk",
     "Junk",
     "Funn",
     "Gunn",
]

for h in fet:
     if h == "Junk":
          break
     print (h)


print ("\n --- batas --- \n")





# Fungsi dasar

def dasar ():
     print ("Hello World")

dasar ()


print ("\n --- batas --- \n")




# Fungsi dengan parameter

def nama (sapa):
     print (f"Halo nama saya {sapa} dari Jakarta Timur")

nama ("Roni")
nama ("Romi")
nama ("Yanzheng")
nama ("Ron")
nama ("Fan")
nama ("Yan")
nama ("Fest")


print ("\n --- batas --- \n")



# Fungsi return

def run (sapa):
     return f"Halo saya {sapa} dari Kota Tegal"

print (run ("Hayyan"))
print (run ("Hafan"))
print (run ("Hun"))
print (run ("Fun"))
print (run ("Der"))


print ("\n --- batas --- \n")


# Fungsi return dengan operator dasar

def tambah (x, y):
     return x + y


def kurang (x, y):
     return x - y


def kali (x, y):
     return x * y


def bagi (x, y):
     return x / y


print ("Hasil tambah =", tambah (10, 9))
print ("Hasil kurang =", kurang (10, 5))
print ("Hasil kali =", kali (10, 10))
print ("Hasil bagi =", bagi (10, 4))


print ("\n --- bataas --- \n")




# Error Handling

try:
     a = 10 + 10
     print (a)

except ZeroDivisionError:
     print ("Gagal")

else:
     print ("Oke")

finally:
     print ("Selesai")


print ("\n --- batas --- \n")




# Error Handling

try:
     b = 10 / 0
     print (b)


except ZeroDivisionError:
     print ("Gagal")

else:
     print ("Oke")

finally:
     print ("Selesai")


print ("\n --- batas --- \n")



# Fungsi dengan percabangan dasar

def der (f):

     if f >= 5:
          print (f"angka f besar, angka f = {f}")

     else:
          print (f"angka f kecil, angka f = {f}")

der (10)
der (9)
der (8)
der (4)
der (3)
der (2)
der (1)


print ("\n --- batas --- \n")



# Fungsi dengan percabangan nested

def erun (e):

     cek = True

     if e >= 5:
          if cek:
               print (f"angka e besar, angka e = {e}")

          else:
               print (f"angka e tengah, angka e = {e}")

     else:
          print (f"angka e kecil, angka e = {e}")

erun (10)
erun (9)
erun (8)
erun (7)
erun (6)
erun (5)
erun (4)
erun (3)
erun (2)
erun (1)


print ("\n --- batas --- \n")




# Fungsi dengan percabangan error handling

def eror (f):

     try:

          if f < 0:
               raise Exception ("Angka minus")

          if f >= 8:
               print (f"angka f besar, angka f = {f}")

          elif f >= 5:
               print (f"angka f tengah, angka f = {f}")

          else:
               print (f"angka f kecil, angka f = {f}")

     except:
          print (f"angka minus, angka f = {f}")

eror (-10)
eror (-44)
eror (-2)
eror (-45)
eror (-5)
eror (10)
eror (9)
eror (5)


print ("\n --- batas --- \n")



# OOP dasar

class Mobil:

     def __init__(self, nama, asal):
          self.nama = nama
          self.asal = asal

     def aksi (self):
          print (f"Mobil {self.nama} dengan asal daerah dari {self.asal}")

hasil_1 = Mobil ("Roy", "Tegal")

hasil_1.aksi ()


print ("\n --- batas --- \n")